cli/cluster/file.go

```diff
@@ -20,7 +23,13 @@ func NewFileCollector(path string) FileCollector {
 
 // Collect collects a configuration from a file located at a specified path.
 func (collector FileCollector) Collect() (*Config, error) {
-	data, err := os.ReadFile(collector.path)
+	file, err := integrity.FileRepository.Read(collector.path)
+	if err != nil {
+		return nil, fmt.Errorf("can't check integrity of %q: %w", collector.path, err)
+	}
+	defer file.Close()
+
+	data, err := io.ReadAll(file)
```

Comment:
I would expect here different implementations in `integrity` module for config collectors/publishers.

This is not so obvious in the context of the file collector, but in other cases the algorithms are too different. As example, you could try to implement `FileDataPublisher` in that way.

Reply:
Why would `FileDataPublisher` be affected in any way by this?

Reply:
Thank you, I see what you've meant now!